### PR TITLE
mingw-w64: update to binutils 2.40

### PR DESCRIPTION
**Pull request by Gcenx**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/mingw-w64.rb [hunk 1]
```diff
@@ -31,9 +31,9 @@ class MingwW64 < Formula
   depends_on "mpfr"
 
   resource "binutils" do
-    url "https://ftp.gnu.org/gnu/binutils/binutils-2.39.tar.xz"
-    mirror "https://ftpmirror.gnu.org/binutils/binutils-2.39.tar.xz"
-    sha256 "645c25f563b8adc0a81dbd6a41cffbf4d37083a382e02d5d3df4f65c09516d00"
+    url "https://ftp.gnu.org/gnu/binutils/binutils-2.40.tar.xz"
+    mirror "https://ftpmirror.gnu.org/binutils/binutils-2.40.tar.xz"
+    sha256 "0f8a4c272d7f17f369ded10a4aca28b8e304828e95526da482b0ccc4dfc9d8e1"
   end
```
Comment (chenrui333) [inline Formula/mingw-w64.rb:37]: instead doing a resource, maybe adding as dependency?

Comment (Porkepix) [reply Formula/mingw-w64.rb:37]: And if so, that could probably also be the case for `gcc`, just below.

Comment (chenrui333) [reply Formula/mingw-w64.rb:37]: yeah, I believe gcc should be a depends_on as well, asking about this. :)

Comment (Gcenx) [reply Formula/mingw-w64.rb:37]: That’s not possible as GNU toolchains don’t support cross-targets unlike an LLVM based toolchain.

We’re forced to compile each component for the desired cross-target.

Comment (chenrui333) [reply Formula/mingw-w64.rb:37]: 👍

**Review by chenrui333 — APPROVED** [review]

**Comment by Gcenx** [comment]
This will lightly still require the following labels `CI-long-timeout` & `long build` this being due to requiring to build gcc multiple times.

**Pull request closed without merging**